src/sinol_make/commands/run/__init__.py

```diff
@@ -0,0 +1,542 @@
+				print_view()
+		if report_file:
+			print_view(report_file)
+		return program_groups_scores
+	
+
+	def validate_subtasks(self):
+		print("Validating subtasks...")
+		if 'subtasks' not in self.config.keys():
+			print(util.bold(util.color_red('Subtasks description not defined in config.yml.')))
+			
+		programs = []
+		for subtask in self.config["subtasks"]:
+			score_checksum = 0
+			for group in  self.config["subtasks"][subtask]["groups"]:
+				if group not in self.scores.keys() or group == 0:
+					print(util.bold(util.color_red('Group %d was not defined.' % group)))
+					exit(1)
+					
+				score_checksum += self.scores[group]
+			score_expected = self.config["subtasks"][subtask]["points"]
+			if score_checksum != score_expected:
+				print(util.bold(util.color_red('Subtask %s will grant %d points (expected %d).' % (subtask, score_checksum, score_expected))))
+				exit(1)
+			
+			validator_program = self.config["subtasks"][subtask]["validator"].split()[0]
+			programs.append(validator_program)
+		programs = list(set(programs))
+		compilation_results = self.compile_programs(programs)
+		os.makedirs(self.EXECUTIONS_DIR, exist_ok=True)
+		compiled_commands = []
+		for subtask in self.config["subtasks"]:
+			validator_program = os.path.join(self.EXECUTABLES_DIR, self.config["subtasks"][subtask]["validator"])
+			compiled_commands.append((subtask, validator_program, True))
+		names = list(self.config["subtasks"])
+		results = self.perform_executions(compiled_commands, names, programs, self.args.subtask_report)
+		for subtask in self.config["subtasks"]:
+			passed_groups = []
+			for group in self.config["scores"]:
+				if results[subtask][group] == "OK":
+					passed_groups.append(group)
+			passed_groups.sort()
+			should_pass = self.config["subtasks"][subtask]["groups"]
+			should_pass.sort()
+			if passed_groups != should_pass:
+				print(util.bold(util.color_red('Subtask %s will pass groups %s (expected %s).' % (subtask, passed_groups, should_pass))))
+				exit(1)
+
+
+	def run_programs(self):
+		programs = self.get_programs(self.args.programs)
+		print("The following %d programs will be executed:\n%s"
+			% (len(programs), [self.extract_program_name(program) for program in programs]))
+		print("on the following %d tests:\n%s"
+			% (len(self.tests), [self.extract_test_no(test) for test in self.tests] ))
+		print("in parallel on %d cpus." % self.cpus)
+		print()
+		compilation_results = self.compile_programs(programs)
+		os.makedirs(self.EXECUTIONS_DIR, exist_ok=True)
+		program_executables = [os.path.join(self.EXECUTABLES_DIR, self.get_executable(program))
+							for program in programs]
+		compiled_commands = zip(programs, program_executables, compilation_results)
+		names = programs
+		self.perform_executions(compiled_commands, names, programs, self.args.program_report)
+	
+	
+	def run(self, args):
+		if not util.check_if_project():
+			print(util.bold(util.color_red('You are not in a project directory.')))
+			exit(1)
```

Comment:
I like that! Let's also add an information (doesn't have to also be in bold/red) why the tool thinks that we are not in a project directory -- e.g. that it didn't find `config.yml` in the current working directory.